docs/advanced/custom-components.md

```diff
@@ -1,84 +1,92 @@
 # Custom Components
 
-[Marker](/docs/api-reference/marker.md),
-[Popup](/docs/api-reference/popup.md), and
-[NavigationControl](/docs/api-reference/navigation-control.md)
-all extend the `BaseControl` React component. You may also create your own map control components.
+You may create your own map control components by consuming [MapContext](/docs/api-reference/map-context.md).
 
-## Example
+## Using the MapContext
 
 The following component renders a label "(longitude, latitude)" at the given coordinate:
 
 ```js
 import * as React from 'react';
-import {BaseControl} from 'react-map-gl';
-
-class CustomMarker extends BaseControl {
-  _render() {
-    const {longitude, latitude} = this.props;
-
-    const [x, y] = this._context.viewport.project([longitude, latitude]);
-
-    const markerStyle = {
-      position: 'absolute',
-      background: '#fff',
-      left: x,
-      top: y
-    };
-
-    return (
-      <div ref={this._containerRef}
-        style={markerStyle} >
-        ({longitude}, {latitude})
-      </div>
-    );
-  }
+import MapGL, {MapContext} from 'react-map-gl';
+
+function CustomMarker(props) {
+  const context = React.useContext(MapContext);
```

Comment:
we could do a further improvement, maybe we can export something like the following:
```
import {useContext} from 'react';
import {MapContext} from '../context';

const useMapContext = () => useContext(FeatureFlagsContext);

export default useMapContext;
```

by doing the following, developers don't need to use `React.useContext(MapContext)` but they can directly use `useMapContext`

Reply:
Is there a significant difference? I personally like `useContext` because the user will be able to refer to React's documentation for the behaviors of context. If we make a custom hook then that's all hidden under the hood.